### PR TITLE
issue #9187 doxygen -x_noenv option to not expand environment variables

### PR DESCRIPTION
**Pull request by albert-github**
Adding the possibility to retain the environment variables (`$(...)`) in the doxyfile when looking for differences with the default settings.

### CODE REVIEW

File: doc/arch.doc [hunk 1]
```diff
@@ -262,7 +262,8 @@ In case one has to communicate through e.g. a forum the configuration settings t
 are different from the standard doxygen configuration file settings one can run the
 doxygen command: with the `-x` option and the name of the configuration file (default
 is `Doxyfile`). The output will be a list of the not default settings (in `Doxyfile`
-format).
+format). Alternatively also `-x_noenv` is possible which is identical to the `-x`
+option but withour replacing the environment variables.
```
Comment (ferdymercury) [inline doc/arch.doc:266]:
```suggestion
option but without replacing the environment variables.
```

File: doc/doxygen.1 [hunk 2]
```diff
@@ -46,6 +46,11 @@ If - is used for extensionsFile doxygen will write to standard output.
 7) Use doxygen to compare the used configuration file with the template configuration file
 .IP
 doxygen \fB\-x\fR [configFile]
+.TP
+   Use doxygen to compare the used configuration file with the template configuration file
+   without replacing the environment variables
+.IP
+doxygen \fB\-x\fR [configFile]
```
Comment (ferdymercury) [inline doc/doxygen.1:53]:
```suggestion
doxygen \fB\-x_noenv\fR [configFile]
```

**Comment by albert-github** [comment]
Some bads from my side, sorry.

**Comment by ferdymercury** [comment]
Thanks so much!

**Pull request merged**